NuclearData/ceNeutronData/ceNeutronMaterial_class.f90

```diff
@@ -215,6 +242,28 @@ elemental function isFissile(self) result(isIt)
 
   end function isFissile
 
+  !!
+  !! Return .true. if TMS is on in the material and the provided energy is not
+  !! within the ures or S(a,b) range in the material
+  !!
+  !! Args:
+  !!   E [in] -> test energy
+  !!
+  !! Result:
+  !!   .TRUE. if conditions for using TMS are satisfied, .FALSE. otherwise
+  !!
+  !! Errors:
+  !!   None
+  !!
+  elemental function useTMS(self, E) result(shouldIt)
```

Comment:
Is there any other OTF Doppler broadening that uses the same machinery as TMS? 

I think I would leave it as is, because this check is used during transport when performing operations that I believe are specific to TMS. Like in ceNeutronMaterial % sampleCollision, there's the TMS route and the normal route. Or when looking up xss, whether you need the total material or the temperature majorant is determined by this check. And I don't how general this really is, but I might be missing something.